### PR TITLE
1904 show algorithm task run data

### PR DESCRIPTION
**Pull request by ramtob**

### CODE REVIEW

File: src/app/@ansyn/menu-items/algorithms/reducers/tasks.reducer.ts [hunk 1]
```diff
@@ -70,6 +105,20 @@ export const selectTaskEntities = <MemoizedSelector<ITasksState, Dictionary<Algo
...
-export const selectAlgorithmTasksSelectedTask: MemoizedSelector<any, any> = createSelector(tasksStateSelector, (algorithmsState: ITasksState) => algorithmsState.selectedTask);
+export const selectAlgorithmTasksPageToShow: MemoizedSelector<any, any> = createSelector(tasksStateSelector, (algorithmsState: ITasksState) => algorithmsState.pageToShow);
+export const selectAlgorithmTasksLoadingFlag: MemoizedSelector<any, any> = createSelector(tasksStateSelector, (algorithmsState: ITasksState) => algorithmsState.loading);
+export const selectAlgorithmTasksSelectedTaskId: MemoizedSelector<any, any> = createSelector(tasksStateSelector, (algorithmsState: ITasksState) => algorithmsState.selectedTaskId);
+export const selectAlgorithmTasksSelectedTask: MemoizedSelector<any, any> = createSelector(tasksStateSelector, (algorithmsState: ITasksState) => {
+	return !algorithmsState.selectedTaskId ?
+		null :
+		selectEntities(algorithmsState)[algorithmsState.selectedTaskId]
+});
+
```
Comment (yairtawil) [inline src/app/@ansyn/menu-items/algorithms/reducers/tasks.reducer.ts]: create selectors with some hierarchy

File: src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.ts [hunk 2]
```diff
@@ -28,62 +44,150 @@ import { ToggleIsPinnedAction } from '@ansyn/menu';
 	destroy: 'ngOnDestroy'
 })
 export class TasksFormComponent implements OnInit, OnDestroy {
-	task: AlgorithmTask = new AlgorithmTask();
-	taskStatus: AlgorithmTaskStatus = 'New';
+	// task: AlgorithmTask = new AlgorithmTask();
 	whichOverlays: AlgorithmTaskWhichOverlays = 'favorite_overlays';
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.ts]: done

File: src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.ts [hunk 3]
```diff
@@ -28,62 +44,150 @@ import { ToggleIsPinnedAction } from '@ansyn/menu';
...
-			}
-		));
+			this.store$.dispatch(new SetCurrentTaskOverlays(overlays || []));
+		})
+	);
 
 	@AutoSubscription
-	activeOverlay$: Observable<IOverlay> = this.store$.pipe(
-		select(mapStateSelector),
+	getMasterOverlayForNewTask$: Observable<IOverlay> = this.isNewTask$.pipe(
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.ts]: done

File: src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.ts [hunk 4]
```diff
@@ -28,62 +44,150 @@ import { ToggleIsPinnedAction } from '@ansyn/menu';
...
 		map((map: ICaseMapState) => map.data.overlay),
+		mergeMap((overlay: IOverlay) => combineLatest(
+			of(overlay),
+			this.store$.select(selectCurrentAlgorithmTaskOverlays)
+			)),
 		distinctUntilChanged(),
-		tap((overlay: IOverlay) => {
-			this.task.state.masterOverlay = overlay;
-			this.checkForErrors();
+		map(([activeOverlay, overlays]: [IOverlay, IOverlay[]]) => {
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/menu-items/algorithms/components/tasks-form/tasks-form.component.ts]: done

File: src/app/@ansyn/menu-items/algorithms/effects/tasks.effects.ts [hunk 5]
```diff
@@ -2,46 +2,52 @@ import { Injectable } from '@angular/core';
...
 import { TasksRemoteService } from '../services/tasks-remote.service';
+import { tap } from 'rxjs/internal/operators';
 
 @Injectable()
 export class TasksEffects {
 
 	@Effect()
 	loadTasks$: Observable<AddTasksAction | {}> = this.actions$.pipe(
-		ofType(AlgorithmsActionTypes.LOAD_TASKS),
+		ofType(TasksActionTypes.LOAD_TASKS),
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/menu-items/algorithms/effects/tasks.effects.ts]: done

File: src/app/@ansyn/menu-items/algorithms/effects/tasks.effects.ts [hunk 6]
```diff
@@ -2,46 +2,52 @@ import { Injectable } from '@angular/core';
...
 		withLatestFrom(this.store.select(selectTaskTotal), (action, total) => total),
+		tap(() => {
+			this.store.dispatch(new SetTasksLoadingFlagAction(true))
+		}),
 		switchMap((total: number) => {
 			return this.tasksService.loadTasks(total).pipe(
+				tap(() => {
+					this.store.dispatch(new SetTasksLoadingFlagAction(false))
+				}),
 				map(tasks => new AddTasksAction(tasks))
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/menu-items/algorithms/effects/tasks.effects.ts]: done

File: src/app/@ansyn/menu-items/algorithms/reducers/tasks.reducer.ts (hunk 1 above)
Comment (ramtob) [reply src/app/@ansyn/menu-items/algorithms/reducers/tasks.reducer.ts]: done

File: src/app/@ansyn/menu-items/algorithms/reducers/tasks.reducer.ts [hunk 7]
```diff
@@ -43,8 +43,12 @@ export function TasksReducer(state: ITasksState = initialTasksState, action: Tas
 			return { ...state, pageToShow: action.payload };
 		}
 
-		case TasksActionTypes.SET_LOADING_FLAG: {
-			return { ...state, loading: action.payload };
+		case TasksActionTypes.LOAD_TASKS: {
```
Comment (yairtawil) [inline src/app/@ansyn/menu-items/algorithms/reducers/tasks.reducer.ts]: Beautiful

**Pull request merged**